[Nandita] iP

nanditaselvaraju opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

nanditaselvaraju committed:
create new main class, add greeting
remove Duke.java

nanditaselvaraju committed:
added code to echo command given by the user

nanditaselvaraju committed:
store tasks in a list, and print out the list of tasks

nanditaselvaraju committed:
ability to mark things as done, and uncheck if not done

nanditaselvaraju committed:
added a ToDo, Deadline and Event class, together with necessary methods

nanditaselvaraju committed:
added a deleteTasks method, implemented I/O testing

LonelyFort commented on text-ui-test/EXPECTED.TXT:
put this file under .gitignore.

LonelyFort commented on text-ui-test/input.txt:
put this file under .gitignore.

LonelyFort commented on src/main/java/Bork.java:
```diff
@@ -0,0 +1,149 @@
+
+public class Bork {
+    private static final int maxTasks = 100;
+    private static ArrayList<Task> tasks = new ArrayList<>();
+    private static int taskCount = 0; // tracking number of tasks
```
If you are using array list you don't need this field. You can just call tasks.size() rather than actively trying to keep track of the number of tasks yourself.

LonelyFort replied:
If you are using array List, I don't think you need the taskCount field? You can call tasks.size() directly without the need to actively keep track of the number of tasks in the list.

LonelyFort reviewed (commented):
Looks good so far, just one minor tweak will do. Best of luck for your week 3 tasks!

chanjunlin commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,27 @@
+
+    public String getStatus() {
+        return (isDone ? "X" : " ");
+    }
```
Perhaps you can make it more descriptive by renaming it to `getCompletionStatus()` to make it clear what the status refers to 😄.

chanjunlin commented on src/main/java/Bork.java:
```diff
@@ -0,0 +1,149 @@
+        }
+    }
+
+    private static void unmarkTask(String[] inputParts) {
```
The method name is descriptive and action-oriented, following the naming conventions, making it easy to understand what this method does 😄. Keep it up!

chanjunlin commented on src/main/java/Bork.java:
```diff
@@ -0,0 +1,149 @@
+        System.out.println("What can I do for you?");
+
+        while (true) {
+            String input = scanner.nextLine().trim();
```
Perhaps renaming it to `userInput` makes it clearer where the input comes from? This could improve readability as the code grows longer 🤔.

chanjunlin commented on src/main/java/Bork.java:
```diff
@@ -0,0 +1,149 @@
+        }
+        scanner.close();
+    }
+    private static void printTaskList() {
```
Good job for choosing a verb for the method's name 😄!

chanjunlin reviewed (commented):
LGTM! Just a few nits to fix 😄 .

nanditaselvaraju committed:
Implement task storage, auto-save and load tasks from disk

nanditaselvaraju committed:
no message

nanditaselvaraju committed:
Update chatbot to use Date, Time format

Code is still not giving out the dates in the right format, still have corrections to be done

nanditaselvaraju committed:
Improve code structure with OOP principles and add ResetCommand
  - Refactored the codebase to follow OOP principles by extracting closely related code into separate classes:
    - Created `Ui` class to handle user interactions.
    - Created `Storage` class to manage file I/O operations.
    - Created `Parser` class to interpret user commands.
    - Created `TaskList` class to manage the list of tasks.
    - Introduced `Command` classes (e.g., `MarkCommand`, `UnmarkCommand`, `AddTodoCommand`, etc.) to encapsulate command logic.
  - Added `ResetCommand` to allow users to reset the task list at the start of the program:
      - Prompts the user with "Do you want to reset the task list? (yes/no): ".
      - Clears the task list and storage file if the user inputs `yes`.
      - Preserves the task list if the user inputs `no`.
  - Ensured backward compatibility with existing functionality while adding new features.

nanditaselvaraju committed:
Revert "Improve code structure with OOP principles and add ResetCommand"

This reverts commit 61bfd89c1dac7e7d02ea565f504513d6bdff499f.

nanditaselvaraju committed:
Refactor: Organise classes into suitable Java packages.

Before organising into packages, there was minimal code maintainability and scalability. To improve the structure, divide the classes into suitable packages:
   - bork.core
   - bork.command
   - bork.task
   - bork.ui
   - bork.storage
   - bork.exception
This refactoring improves code organization, adheres to OOP principles, and makes the project more maintainable and scalable.

nanditaselvaraju committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

nanditaselvaraju committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Refactor: Organise classes into suitable Java packages.
  Improve code structure with OOP principles and add ResetCommand   - Refactored the codebase to follow OOP principles by extracting closely related code into separate classes:     - Created `Ui` class to handle user interactions.     - Created `Storage` class to manage file I/O operations.     - Created `Parser` class to interpret user commands.     - Created `TaskList` class to manage the list of tasks.     - Introduced `Command` classes (e.g., `MarkCommand`, `UnmarkCommand`, `AddTodoCommand`, etc.) to encapsulate command logic.   - Added `ResetCommand` to allow users to reset the task list at the start of the program:       - Prompts the user with "Do you want to reset the task list? (yes/no): ".       - Clears the task list and storage file if the user inputs `yes`.       - Preserves the task list if the user inputs `no`.   - Ensured backward compatibility with existing functionality while adding new features.
  Update chatbot to use Date, Time format

nanditaselvaraju committed:
Test: Add JUnit tests for TaskList and Storage classes

To ensure reliability of the code, add JUnit tests for two non-trivial methods:
   1. TaskList: Tested the add and remove methods to verify task management functionality
   2. Storage: Tested the save and load methods to ensure that tasks are correctly persisted and retrieved

nanditaselvaraju committed:
Add JavaDoc comments to improve code documentation

Adding JavaDoc comments enhances code readability and maintainability by providing clear explanations of classes, methods, and parameters. This improves understanding for developers working on the codebase and ensures consistency in documentation.

Let's ensure that all public methods include meaningful descriptions, parameter details, and return values where applicable.

nanditaselvaraju committed:
Feature: Add find command to search tasks by keyword

Users can now search for tasks containing a specific keyword in their descriptions using the find command.
The implementation includes:
   - A new FindCommand class to handle the search logic.
   - Updates to the Parser class to recognise the find command.
   - A new method in the Ui class to display matching tasks.
This feature improves usability by allowing users to quickly locate tasks based on keywords.

nanditaselvaraju committed:
Fix coding style violations and add Javadoc comments

- Added missing Javadoc comments to all classes and methods.
- Reordered imports to follow lexicographical order as per Checkstyle rules.
- Fixed indentation issues in switch cases and method blocks.
- Replaced wildcard imports with explicit static imports in test classes.
- Wrapped long lines to adhere to the 120-character limit.
- Ensured proper spacing around keywords (e.g., if statements).

Checkstyle now passes without errors.

nanditaselvaraju committed:
💻 Add JavaFX GUI for chatbot interaction

- Implemented JavaFX-based GUI for the chatbot.
- Followed SE-EDU JavaFX tutorial up to Part 4, and changed the colours of the backgrounds and labels.
- Ensured a separate Launcher class for compatibility.
- Users can now interact with the chatbot through the graphical interface. 🚀

nanditaselvaraju committed:
Fix GUI response handling and display welcome message on start up

- Fixed issue where GUI was mirroring user input instead of processing it through Bork.
- Ensured that the welcome message is displayed automatically when the program starts.

nanditaselvaraju committed:
Feature: Add assertions to enforce critical assumptions in ToDo class

Use Java's assert feature to document and enforce key assumptions in the ToDo class. These assertions help catch potential issues early and ensure data integrity.
- Add an assertion in the constructor to verify that the task description is not null or empty.
- Include assertions in `toFileString()` and `toString()` methods  to ensure that the description is not null before processing.
- Improve overall code robustness by validating essential conditions at runtime during development.

These assertions help prevent unexpected behaviour and improve code reliability.

nanditaselvaraju committed:
Refactor: Task class to improve code quality and maintainability

- Enforced encapsulation by making 'description' private final.
- Introduced a reusable DateTimeFormatter to avoid duplication.
- Improved error handling in fromFileString() by throwing IllegalArgumentException instead of returning null.
- Extracted helper methods (parseDeadline, parseEvent) to improve readability and maintainability.
- Followed Code Quality principles for better structure and single responsibility.

This refactor ensures better error handling, readability, and future extensibility.

nanditaselvaraju committed:
Merge pull request #3 from nanditaselvaraju/branch-A-Assertions

Feature: Add assertions to enforce critical assumptions in ToDo class

nanditaselvaraju committed:
Merge branch 'master' into branch-A-CodeQuality

nanditaselvaraju committed:
Merge pull request #2 from nanditaselvaraju/branch-A-CodeQuality

Refactor: Task class to improve code quality and maintainability

nanditaselvaraju committed:
feat: implement C-Sort extension to sort tasks chronologically

- Add `SortCommand` to handle the sorting of tasks
- Update `UserInterface` with `showSortedTasks` method to display sorted tasks
- Modify `TaskList` to include `sortTasks` method for chronological sorting
- Update `Parser` to recognise the "sort" command

This feature allows users to sort their tasks in chronological order, improving task organization and accessibility.

nanditaselvaraju committed:
Add files via upload

nanditaselvaraju committed:
Update README.md
